map/gps_track.cpp

```diff
@@ -351,3 +359,27 @@ void GpsTrack::NotifyCallback(pair<size_t, size_t> const & addedIds, pair<size_t
     m_callback(std::move(toAdd), evictedIds);
   }
 }
+
+std::vector<location::GpsTrackInfo> GpsTrack::GetCurrentTrack() const
+{
+  lock_guard<mutex> lg(m_dataGuard);
```

Comment:
Since C++17 the type should not be needed.
```suggestion
  std::lock_guard lg(m_dataGuard);
```